tools/serverpod_cli/test/analyzer/entities/entity_analyzer/relation_one_to_one_test.dart

```diff
@@ -0,0 +1,536 @@
+  oldAddressId: int
+  oldAddress: Address?, relation(name=company_address, field=oldAddressId)
+''',
+      Uri(path: 'lib/src/protocol/address.yaml'),
+      [],
+    );
+
+    var protocol2 = ProtocolSource(
+      '''
+class: Address
+table: address
+fields:
+  company: Company?, relation(name=company_address)
+''',
+      Uri(path: 'lib/src/protocol/company.yaml'),
+      [],
+    );
+
+    var definition1 = SerializableEntityAnalyzer.extractEntityDefinition(
+      protocol1,
+    );
+
+    var definition2 = SerializableEntityAnalyzer.extractEntityDefinition(
+      protocol2,
+    );
+
+    var entities = [definition1!, definition2!];
+
+    SerializableEntityAnalyzer.resolveEntityDependencies(entities);
+
+    SerializableEntityAnalyzer.validateYamlDefinition(
+      protocol1.yaml,
+      protocol1.yamlSourceUri.path,
+      collector,
+      definition1,
+      entities,
+    );
+    SerializableEntityAnalyzer.validateYamlDefinition(
+      protocol2.yaml,
+      protocol2.yamlSourceUri.path,
+      collector,
+      definition2,
+      entities,
+    );
+
+    expect(
+      collector.errors,
+      isNotEmpty,
+      reason: 'Expected an error but none was found.',
+    );
+
+    var error = collector.errors.first;
+
+    expect(
+      error.message,
+      'Unable to resolve ambiguous relation, there are several named relations with name "company_address" on the class "Company".',
+    );
+  });
+
+  test(
+      'Given a class with a one to one relation where the relationship is ambiguous where the field names are the same in both models then an error is collected that the reference cannot be resolved.',
+      () {
+    var collector = CodeGenerationCollector();
+
+    var protocol1 = ProtocolSource(
+      '''
+class: Company
+table: company
+fields:
+  addressCompany: int
+  address: Address?, relation(name=company_address, field=addressCompany)
+  oldAddressId: int
+  oldAddress: Address?, relation(name=company_address, field=oldAddressId)
+''',
+      Uri(path: 'lib/src/protocol/address.yaml'),
+      [],
+    );
+
+    var protocol2 = ProtocolSource(
+      '''
+class: Address
+table: address
+fields:
+  addressCompany: Company?, relation(name=company_address)
+''',
+      Uri(path: 'lib/src/protocol/company.yaml'),
+      [],
+    );
+
+    var definition1 = SerializableEntityAnalyzer.extractEntityDefinition(
+      protocol1,
+    );
+
+    var definition2 = SerializableEntityAnalyzer.extractEntityDefinition(
+      protocol2,
+    );
+
+    var entities = [definition1!, definition2!];
+
+    SerializableEntityAnalyzer.resolveEntityDependencies(entities);
+
+    SerializableEntityAnalyzer.validateYamlDefinition(
+      protocol1.yaml,
+      protocol1.yamlSourceUri.path,
+      collector,
+      definition1,
+      entities,
+    );
+    SerializableEntityAnalyzer.validateYamlDefinition(
+      protocol2.yaml,
+      protocol2.yamlSourceUri.path,
+      collector,
+      definition2,
+      entities,
+    );
```

Comment:
Suggestion: Move this into a helper method please 🙏 

After reading a bunch of these tests this boilerplate just makes the test harder to understand. It separates the tests cases from the definitions that are being validated. I don't think it helps with the understanding of the test and doesn't really reflect the abstraction level we are testing on.

```suggestion
    var definition1 = SerializableEntityAnalyzer.extractEntityDefinition(
      protocol1,
    );

    var definition2 = SerializableEntityAnalyzer.extractEntityDefinition(
      protocol2,
    );

    var entities = [definition1!, definition2!];

    SerializableEntityAnalyzer.resolveEntityDependencies(entities);

    SerializableEntityAnalyzer.validateYamlDefinition(
      protocol1.yaml,
      protocol1.yamlSourceUri.path,
      collector,
      definition1,
      entities,
    );
    SerializableEntityAnalyzer.validateYamlDefinition(
      protocol2.yaml,
      protocol2.yamlSourceUri.path,
      collector,
      definition2,
      entities,
    );
```

Reply:
Will do in another PR